masonry/examples/grid_masonry.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This is a very small example of how to setup a masonry application.
+//! It does the almost bare minimum while still being useful.
+
+// On Windows platform, don't show a console when opening the app.
+#![windows_subsystem = "windows"]
+
+use masonry::app_driver::{AppDriver, DriverCtx};
+use masonry::dpi::LogicalSize;
+use masonry::widget::{Button, Grid, GridParams, Label, RootWidget};
+use masonry::{Action, PointerButton, WidgetId};
+use winit::window::Window;
+
+struct Driver {
+    grid_spacing: f64,
+}
+
+impl AppDriver for Driver {
+    fn on_action(&mut self, ctx: &mut DriverCtx<'_>, _widget_id: WidgetId, action: Action) {
+        match action {
+            Action::ButtonPressed(button) => {
+                if button == PointerButton::Primary {
+                    self.grid_spacing += 1.0;
+                } else if button == PointerButton::Secondary {
+                    self.grid_spacing -= 1.0;
+                } else {
+                    self.grid_spacing += 0.5;
+                }
+
+                ctx.get_root::<RootWidget<Grid>>()
+                    .get_element().set_spacing(self.grid_spacing)
+            }
+            _ => ()
+        }
+    }
+}
+
+struct DrawnButton {
+    grid_params: GridParams,
+}
+
+impl DrawnButton {
+    fn get_label(&self) -> String {
+        format!("X: {}, Y: {}, W: {}, H: {}", self.grid_params.x, self.grid_params.y, self.grid_params.width, self.grid_params.height)
+    }
+}
+
+pub fn main() {
+    let label = Label::new("Change spacing by right and\n left clicking on the buttons")
```

Comment:
```suggestion
    let label = Prose::new("Change spacing by right and\n left clicking on the buttons")
```
Labels should generally only be used for text inside other interactive widgets.